aws-android-sdk-testutils/src/main/java/com/amazonaws/testutils/util/InternetConnectivity.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ * http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazonaws.testutils.util;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.net.Socket;
+import java.util.concurrent.TimeUnit;
+
+public final class InternetConnectivity {
+    private InternetConnectivity() {}
+
+    public static void goOnline() {
```

Comment:
do we need to synchronize these methods? Can there be a race condition between checking for connection status vs changing connection status?

Reply:
It couldn't hurt. I'll add a big ol'
```java
synchronized (InternetConnectivity.class) {
}
```
around it.